Add release notes for 5.20 (#512)
chef/cookstyle: RELEASE_NOTES.md

## Cookstyle 5.20

### 1 New Cop

#### ChefDeprecations/DeprecatedChefSpecPlatform

> The `ChefDeprecations/DeprecatedChefSpecPlatform` cop detects the usage of platforms in ChefSpec tests that have been deprecated in the Fauxhai / ChefSpec projects. Using these older platforms will cause you specs to error as mock Ohai data cannot be loaded to emulate a full Chef Infra Client run.

Reviewer: ```suggestion
The `ChefDeprecations/DeprecatedChefSpecPlatform` cop detects platforms in ChefSpec tests that have been deprecated in the Fauxhai / ChefSpec projects. Specifying deprecated platforms will create errors in ChefSpec because the mock Ohai data cannot be loaded to emulate a full Chef Infra Client run.
```

> These deprecations occur as the Fauxhai project, which provides the mock Ohai data, has to be pruned from time to time to avoid the package size from becoming too large. The solution to avoid this problem in your specs is to utilize the newer platform matching functionality in ChefSpec. Previously when writing specs you needed to specify the exact platform release, but now you can skip the version to let ChefSpec load the latest or you can provide just the major release and let ChefSpec do the rest.

Reviewer: ```suggestion
Platforms are deprecated from the Fauxhai project from time to time to prevent the Fauxhai package size from becoming too large. To avoid specifying deprecated platforms, use the newer platform matching functionality in ChefSpec. Previously you needed to specify the exact platform release when writing specs, but now ChefSpec will load the latest version if you skip the version number, or you can provide the major release and let ChefSpec do the rest.
```

`Examples`

ChefSpec calling a deprecated release of Debian 8:

```ruby
let(:chef_run) { ChefSpec::ServerRunner.new(platform: 'debian', version: '8.2') }
```

> ChefSpec calling using the latest minor version of Debian 8 instead:

Reviewer: ```suggestion
ChefSpec calling the latest minor version of Debian 8 instead:
```

```ruby
let(:chef_run) { ChefSpec::ServerRunner.new(platform: 'debian', version: '8') }
```

> ChefSpec calling using the latest major.minor version of Debian instead:

Reviewer: Should this just be major?

```suggestion
ChefSpec calling using the latest major version of Debian instead:
```

Author: I think this one is correct as-is. It's not just the latest major. It's the lastest major.minor version. 8.whatever

```ruby
let(:chef_run) { ChefSpec::ServerRunner.new(platform: 'debian') }
```

### Other fixes and changes

- The `ChefModernize/ExecuteSysctl` cop will now detect `execute` resources that call systctl using the full path to the binary.

> - The `ChefModernize/LibarchiveFileResource` resource will now autocorrect usage of the `libarchive_file` resource to use Chef Infra Client's built-in `archive_file` resource.

Reviewer: ```suggestion
- The `ChefModernize/LibarchiveFileResource` resource will now replace the `libarchive_file` resource with Chef Infra Client's built-in `archive_file` resource.
```

- Various cops that remove legacy code no longer leave behind empty lines when autocorrecting.
- `ChefRedundantCode/LongDescriptionMetadata` now properly autocorrects when `long_description` uses a here document (heredoc).
- `ChefDeprecations/UserDeprecatedSupportsProperty` no longer fails to autocorrect when the supports hash uses Ruby 1.8 hash rocket syntax.

> - `ChefStyle/FileMode` now autocorrects file modes using single quotes instead of double-quotes.

Reviewer: ```suggestion
- `ChefStyle/FileMode` now autocorrects file modes using single quotes instead of double quotes.
```
